### PR TITLE
Improve V2 compatibility

### PR DESCRIPTION
**Pull request by joelverhagen**
1. Detect supported frameworks from package entries as well as framework assembly groups.
2. Support the `ignoreFilter` parameter on V2 search which allows the search hijacker in the gallery to find unlisted packages.

@maartenba @joyhui @xavierdecoster @johnataylor

### CODE REVIEW

File: src/Ng/packages.config [hunk 1]
```diff
@@ -11,7 +11,7 @@
   <package id="Microsoft.Web.Xdt" version="2.1.1" targetFramework="net45" />
   <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.1.0" targetFramework="net45" />
   <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
-  <package id="NuGet.Core" version="2.8.5" targetFramework="net45" />
+  <package id="NuGet.Core" version="2.10.1" targetFramework="net45" />
```
Comment (joelverhagen) [inline src/Ng/packages.config]: It seems like there is a general dislike of NuGet.Core -- is there a newer alternative I should be using for parsing target frameworks? Perhaps I can have a seperate PR as switching away from NuGet.Core like [Maarten's PR in the gallery](https://github.com/NuGet/NuGetGallery/pull/2783)?

File: NgTests/app.config [hunk 2]
```diff
@@ -22,6 +22,10 @@
         <assemblyIdentity name="Microsoft.WindowsAzure.Storage" publicKeyToken="31bf3856ad364e35" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-4.3.0.0" newVersion="4.3.0.0" />
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="NuGet.Core" publicKeyToken="31bf3856ad364e35" culture="neutral" />
```
Comment (yishaigalatzer) [inline NgTests/app.config]: I suggest you don't take new dependencies on nuget.core

Comment (joelverhagen) [inline NgTests/app.config]: Sure, I'll look into switching this over.

File: src/Ng/packages.config (hunk 1 above)
Comment (maartenba) [inline src/Ng/packages.config]: Could be good, I believe the plan is to deprecate NuGet.Core somewhere in the future so being on the new packages certainly won't hurt.

**Comment by maartenba** [comment]
:shipit:

File: src/Ng/packages.config [hunk 3]
```diff
@@ -11,7 +11,7 @@
...
   <package id="nuget.versioning" version="3.2.0" targetFramework="net45" />
   <package id="SharpZipLib" version="0.86.0" targetFramework="net45" />
```
Comment (yishaigalatzer) [inline src/Ng/packages.config]: Consider moving the project off packages.config to project.json (still with csproj) that should work if the project doesn't use any content

File: src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs [hunk 4]
```diff
@@ -189,29 +195,44 @@ private void AddSupportedFrameworks()
...
-                                builder.Append("|");
-                            }
-
-                            builder.Append(shortFrameworkName);
-                        }
+                        yield return VersionUtility.ParseFrameworkName(frameworkName);
                     }
                 }
+            }
+            private IEnumerable<FrameworkName> GetFrameworksFromPaths()
```
Comment (yishaigalatzer) [inline src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs]: nit: new line between the two methods

File: NgTests/app.config (hunk 2 above)
Comment (joelverhagen) [inline NgTests/app.config]: Fixed.

File: src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs (hunk 4 above)
Comment (joelverhagen) [inline src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs]: Fixed.

File: src/Ng/packages.config (hunk 3 above)
Comment (joelverhagen) [inline src/Ng/packages.config]: Out of scope for this PR. Tracked by NuGet/NuGetGallery#2813.

**Pull request merged**